lib/redis/commands/sorted_sets.rb

```diff
@@ -167,6 +167,41 @@ def zpopmin(key, count = nil)
         end
       end
 
+      # Removes and returns up to count members with scores in the sorted set stored at key.
+      #
+      # @example Popping a member
+      #   redis.zmpop('zset')
+      #   #=> ['zset', ['a', 1.0]]
+      # @example With count option
+      #   redis.zmpop('zset', count: 2)
+      #   #=> ['zset', [['a', 1.0], ['b', 2.0]]
+      #
+      # @params key [String, Array] a key of the sorted set(s)
+      # @params min_or_max [String] to pop from min or max
+      # @params count [Integer] a number of members
+      #
+      # @return [Array<String, Array<String, Float>>] list of popped elements and scores
+      def zmpop(*keys, min_or_max: "MIN", count: nil)
+        raise ArgumentError, "Pick either MIN or MAX" unless min_or_max.match?(/MAX|MIN/)
+
+        number_of_keys = keys.is_a?(Array) ? keys.size : 1
+        args = [:zmpop, number_of_keys, *keys, min_or_max]
+
+        if count
+          args << "COUNT"
+          args << Integer(count)
+        end
+
+        send_command(args) do |response|
+          response&.map do |entry|
+            case entry
+            when String then entry
+            when Array then entry.map { |pair| FloatifyPairs.call(pair) }.flatten(1)
+            end
+          end
+        end
+      end
```

Comment:
```suggestion
      # Removes and returns up to count members with scores from the first non-empty sorted-set in the provided list of key names.
      #
      # @example Popping a member
      #   redis.zmpop('zset')
      #   #=> ['zset', ['a', 1.0]]
      # @example With count option
      #   redis.zmpop('zset', count: 2)
      #   #=> ['zset', [['a', 1.0], ['b', 2.0]]
      #
      # @params key [String, Array<String>] one or more keys with sorted sets
      # @params modifier [String]
      #  - when `"MIN"` - the elements popped are those with the lowest scores
      #  - when `"MAX"` - the elements popped are those with the highest scores
      # @params count [Integer] a number of members to pop
      #
      # @return [Array<String, Array<String, Float>>] list of popped elements and scores
      def zmpop(*keys, modifier: "MIN", count: 1)
        raise ArgumentError, "Pick either MIN or MAX" unless modifier == "MIN" || modifier == "MAX"

        args = [:zmpop, keys.size, *keys, modifier]

        if count
          args << "COUNT"
          args << Integer(count)
        end

        send_command(args) do |response|
          response&.map do |entry|
            case entry
            when String then entry
            when Array then FloatifyPairs.call(entry)
            end
          end
        end
      end
```

Reply:
I included most of these suggestions. Your line 235 does not work:

```ruby
irb(main):005:0> redis.zmpop("myzset", modifier: "MAX", count: 2)
=> ["myzset", [[["three", "3"], ["two", "2"]]]]
```

Reply:
Thanks. I haven't tested it before suggesting, so I was thinking that it should work 😎

Reply:
You forgot to change the command description to the suggested (looks more clear imo) and to remove `number_of_keys` variable - `args` is already an array (because of `*args`).

Reply:
Updated, appreciate it.